### PR TITLE
Add accessible attributes to notifications

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

Add accessible attributes to `Notification` component.

#### Description

Currently we are not using any attribute in `Notification` component that can help assistive technologies to properly be aware of this kind of elements when rendering them in the DOM.

The idea is to provide a proper _aria_ role and related attributes following [the spec](https://www.w3.org/TR/wai-aria-1.1/#alertdialog).

_(This is part of what we intend to work on related to #2059)_

### CODE REVIEW

File: packages/react-notifications/src/components/notification/notification.tsx [hunk 1]
```diff
@@ -25,6 +27,8 @@ import {
   getStylesForNotification,
 } from './notification.styles';
 
+const sequentialId = createSequentialId('notification-');
```
Comment (emmenko) [inline packages/react-notifications/src/components/notification/notification.tsx:30]: FYI: for us to aware.

With React v18 there is a new hook `useId` that we should be able to use as a replacement for our custom implementation.

https://reactjs.org/blog/2022/03/29/react-v18.html#useid

**Review by emmenko — APPROVED** [review]
Nice! 💯

File: packages/react-notifications/src/components/notification/notification.tsx (hunk 1 above)
Comment (CarlosCortizasCT) [reply packages/react-notifications/src/components/notification/notification.tsx:30]: Our id generators was the first thing that came to my mind when I read about that new hook 😄

**Review by kark — APPROVED** [review]
Spot on 🙌

**Review by ahmehri — APPROVED** [review]
💯

**Pull request merged**